### PR TITLE
Performance audit

### PR DESCRIPTION
**Pull request by katarighe**
In this activity, we have identified and mitigated the performance issues in the Vet Clinic database queries by implementing optimization strategies. The database performance audit revealed certain queries that were taking longer than desired to execute. 

This work has been done by me (@katarighe) in conjunction with @noelfoka. 

To address this, we applied the following optimizations:
- **Optimized** the `SELECT COUNT(*)` query for visits by adding an index on the `animal_id` column.
- **Improved** the `SELECT * FROM` visits query for vet lookups by adding an index on the `vet_id` column.
- **Enhanced** the `SELECT * FROM` owners query for email-based lookups by introducing an index on the email column.

The following changes have been made aside from the optimizations:
- **Added** appropriate indexes to the relevant columns based on query patterns.
- **Rewrote** queries to effectively utilize indexes and minimize unnecessary data retrieval.
- **Ran** `EXPLAIN ANALYZE` to validate performance improvements after optimization.

The screenshots in this activity are listed as follows:

# Preparation 
**Added an extra column to the owners table.**
image

**Created rows for the `visits_animal_id`, `visits_vet_id`, and `owners_email`.** 
image

**Run the following statements to add data to my database.**
image

**Ensured that the database speed is considered enough for the round of activities.**
image

# Optimization & Performance Audit
**Before**
image
image
image

**After**
image
image
image

### CODE REVIEW

File: data.sql
Comment (AbelG101) [inline data.sql:1]: - [x] Ouch! this is a close one but the execution time in the screenshots for the first and last queries BEFORE is significantly lower than 1 sec (1000ms) as such I'd suggest running the populate database query a few times more as described in the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_performance_audit.md#:~:text=4%20Depening%20on,bigger%20than%201000ms.) to ensure the queries before indexing should be around 1 sec (1000 ms). After indexing the change should be significantly lower than 1sec.
- These are the queries with small execution time before indexing 
  270319123-cd64fcc2-2fd2-43a1-b29f-8d18c9eda5fb
  270320816-9849de13-b320-4e74-a304-f9891f804853

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @katarighe & @noelfoka 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Correct use of `EXPLAIN ANALYZE` ✔
- Proper table indexing ✔
- Correct use of Git Flow ✔
- Descriptive PR title with a great screenshot summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql
Comment (katarighe) [reply data.sql:1]: Hi @AbelG101 

Here are the corrections for these queries:
image
image

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @katarighe & @noelfoka, 👋 👋  

Even if the execution times difference is very small in your case, I can clearly see that you had some fun with the explain analyze and I am sure that in the future you will know how to start the investigation for some slow DB queries :)

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Queries Screen added to the PR ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by katarighe** [comment]
Thanks @brainconnect93, 

I agree that I will learn more of these tasks in the future and I will consider this when investigating slow DBs.

Since I am familiar with databases, that one looks quite confusing to me since it is the first time that I am encountering this task, but this will be considered a lesson for me. 😊🙏

**Pull request merged**